modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java

```diff
@@ -52,866 +54,960 @@
+                intent.putExtra(resources.getString(R.string.intent_extra_message), message);
+                intent.putExtra(resources.getString(R.string.intent_extra_type),
+                        resources.getString(R.string.intent_extra_alert));
+                intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP |
+                        Intent.FLAG_ACTIVITY_NEW_TASK);
+                context.startActivity(intent);
+
+                if (Constants.DEBUG_MODE_ENABLED) {
+                    Log.d(TAG, "Notification received");
+                }
+            }
+        } catch (JSONException e) {
+            operation.setStatus(resources.getString(R.string.operation_value_error));
+            resultBuilder.build(operation);
+            throw new AndroidAgentException("Invalid JSON format.", e);
+        }
+    }
+
+    /**
+     * Configure device WIFI profile.
+     *
+     * @param operation - Operation object.
+     */
+    public void configureWifi(org.wso2.mdm.agent.beans.Operation operation) throws AndroidAgentException {
+        boolean wifiStatus;
+        String ssid = null;
+        String password = null;
+        JSONObject result = new JSONObject();
+
+        try {
+            JSONObject wifiData = new JSONObject(operation.getPayLoad().toString());
+            if (!wifiData.isNull(resources.getString(R.string.intent_extra_ssid))) {
+                ssid = (String) wifiData.get(resources.getString(R.string.intent_extra_ssid));
+            }
+            if (!wifiData.isNull(resources.getString(R.string.intent_extra_password))) {
+                password =
+                        (String) wifiData.get(resources.getString(R.string.intent_extra_password));
+            }
+        } catch (JSONException e) {
+            operation.setStatus(resources.getString(R.string.operation_value_error));
+            resultBuilder.build(operation);
+            throw new AndroidAgentException("Invalid JSON format.", e);
+        }
+
+        WiFiConfig config = new WiFiConfig(context.getApplicationContext());
+
+        wifiStatus = config.saveWEPConfig(ssid, password);
+
+        try {
+            String status;
+            if (wifiStatus) {
+                status = resources.getString(R.string.shared_pref_default_status);
+                result.put(resources.getString(R.string.operation_status), status);
+
+            } else {
+                status = resources.getString(R.string.shared_pref_false_status);
+                result.put(resources.getString(R.string.operation_status), status);
+            }
+        } catch (JSONException e) {
+            operation.setStatus(resources.getString(R.string.operation_value_error));
+            resultBuilder.build(operation);
+            throw new AndroidAgentException("Invalid JSON format.", e);
+        }
+        if (Constants.DEBUG_MODE_ENABLED) {
+            Log.d(TAG, "Wifi configured");
+        }
```

Comment:
At this point, operation is almost completed and applied to the device as well. That is why debug log is set here.